### PR TITLE
fix(checkbox, radio): add a default id when id prop isn't passed

### PR DESCRIPTION
**Pull request by shleewhite**
Currently, id is not a required prop for checkbox or radio, but if you don't pass it the control doesn't work.

This was found in this issue: https://github.com/twilio-labs/paste/issues/2528#issuecomment-1224286228

## In this PR
For both checkbox and radio:
* if the id prop isn't set, use a `UID`
* add a story where the id is not set
* add a test to ensure there is a default id and that the id prop is working as expected

## Why this fixes it
The way our checkboxes and radios work is that the actual control is visually hidden and the stuff we see is all inside the label. That way we can rely on the default browser behavior where clicking the label focuses the control.

Before, when the id prop wasn't passed, the label and the hidden control weren't actually connected anymore. So clicking the visual checkbox/label didn't focus the hidden control to update the state.

Now, there is a default id so even if the prop isn't passed, the label and the hidden checkbox are always connected in the HTML.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by gloriliale — COMMENTED** [review]
Great fix! Just left a comment about testing- should be a v easy fix.

File: packages/paste-core/components/radio-group/__tests__/radio.test.tsx [hunk 1]
```diff
@@ -37,8 +37,22 @@ const defaultGroupProps = {
 
 describe('Radio', () => {
   it('should render', () => {
-    const {getByRole} = render(<Radio {...defaultProps}>foo</Radio>);
-    expect(getByRole('radio')).not.toBeNull();
+    render(<Radio {...defaultProps}>foo</Radio>);
+
+    const radio = screen.getByRole('radio');
+    expect(radio).not.toBeNull();
+    expect(radio.id).toBe('foo');
```
[the remark this answers is not in the document]
Comment (shleewhite) [reply packages/paste-core/components/radio-group/__tests__/radio.test.tsx:44]: Ah, it actually does have the id 'foo'- it is spread in the default props object. You caught me being lazy 😂 , I'll swap the tests around so its like checkbox.

**Review by gloriliale — APPROVED** [review]
nice!

**Review by SiTaggart — APPROVED** [review]
Looks good. I checked all the stories, controlled and otherwise and nothing seems to have broken. So I'm calling it good.

**Pull request merged**